error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/StringRules.java

```diff
@@ -244,4 +244,105 @@ int after(String str) {
       return Utf8.encodedLength(str);
     }
   }
+
+  /** Prefer {@link String#indexOf(int, int)} over less efficient alternatives. */
+  static final class StringIndexOfChar {
```

Comment:
Here and below: these rules will pose a nice challenge once we start to generate rule naming suggestions. For now I kept it simple.